hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/validator/SparkPreCommitValidator.java

```diff
@@ -73,7 +79,23 @@ public void validate(String instantTime, HoodieWriteMetadata<O> writeResult, Dat
     try {
       validateRecordsBeforeAndAfter(before, after, getPartitionsModified(writeResult));
     } finally {
-      LOG.info(getClass() + " validator took " + timer.endTimer() + " ms");
+      long duration = timer.endTimer();
+      LOG.info(getClass() + " validator took " + duration + " ms" + ", metrics on? " + getWriteConfig().isMetricsOn());
+      publishRunStats(instantTime, duration);
+    }
+  }
+
+  /**
+   * Publish pre-commit validator run stats for a given commit action.
+   */
+  private void publishRunStats(String instantTime, long duration) {
+    // Record validator duration metrics.
+    if (getWriteConfig().isMetricsOn()) {
+      HoodieTableMetaClient metaClient = getHoodieTable().getMetaClient();
+      Option<HoodieInstant> currentInstant = metaClient.getActiveTimeline()
+          .findInstantsAfterOrEquals(instantTime, 1)
+          .firstInstant();
+      metrics.reportMetrics(currentInstant.get().getAction(), getClass().getSimpleName(), duration);
```

Comment:
not too strong on the suggestion. 
but we can fetch the write operation type from HoodieCommitMetadata which is within HoodieWriteMetadata. 
we can avoid loading the timeline if we go that route.